R/annotate_intrachromosomal_sv.R

```diff
@@ -0,0 +1,124 @@
+#' @title Annotate intrachromosomal SV.
+#'
+#' @description Annotate the intrachromosomal events in a data frame of SV breakpoints.
+#'
+#' @details Specify a data frame with SVs (preferably the output from get_manta_sv() to the `sv_data` parameter and get back the same data frame with SV annotations.
+#'
+#' @param sv_data A data frame of SVs. This should be the output of get_manta_sv.
+#' The required columns are columns for both chromosomes, coordinates and strand plus SOMATIC_SCORE and tumour_sample_id.
+#' @param return_as Stated format for returned output, default is "bedpe". Other accepted output formats are "bed" and "bedpe_entrez" (to keep entrez_ids for compatibility with portal.R and cBioPortal).
+#' @param projection Reference genome build parameter. Can be one of grch37 or hg38. The default is grch37.
+#' @param types A vector of sv types to be annotated. The default is set to "DEL", "DUP", "INS", and "INV".
+#' @param max_size An integer specifying the maximum distance between start of first breakpoint and end of second breakpoint. The default is 100000 (100000 bp).
+#' @param keep_genes Optional argument allowing ot only return svs annotated to the list of genes provided in this argument. The default is "all" (return all svs annotated to all genes).
+#' @param verbose Optionally control whether to print any status info to console. Useful for debugging. Default is FALSE (no messages).
+#'
+#'
+#' @return A data frame with annotated SVs (gene symbol and entrez ID).
+#'
+#' @rawNamespace import(data.table, except = c("last", "first", "between", "transpose"))
+#' @import tidyr dplyr stringr
+#' @export
+#'
+#' @examples
+#' sv_df = get_manta_sv(verbose = FALSE)
```

Comment:
Thanks for updating this. I have a thought and I am interested in hearing your input. The updated example works fine. However, since the example is giving all manta SVs to the function, the example takes a very long time to run. Maybe this will cause issues, or at least a minor inconvenience when `devtools::check()` or the **Git Action** is run in future PRs? Perhaps it would make more sense to provide an example using just one sample ID? The downside is that the selected sample ID would have to match what's bundled in the GAMBLR.data bedpe (DOHH2 perhaps?), but then again, the bundled SVs are probably not sufficient to make any sense of this function.

Reply:
Yeah I updated it now to be one-sample-only and did the same for the maf example